### ISSUE TITLE
[Bug][a11y]: MenuItem lacks role and state state attributes in docs

### ISSUE THREAD

**Issue by majornista** [adobe/spectrum-web-components#3247]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

Menu, MenuItem, MenuGroup

### Expected behavior

Within a Menu or MenuGroup, a MenuItem should have the appropriate `role` and `aria-checked` depending on the `selects` of the ancestor Menu or MenuGroup, `role="menuitemradio"` with `selects="single"`, `role="menuitemcheckbox"` with `selects="multiple"`, or `role="menuitem"` when `selects` is `undefined`. 

Also, if a MenuItem opens a submenu, it should include `aria-haspopup="true"`, `aria-expanded` to indicate the current state, and `aria-controls` with the IDREF for the expanded popup menu.

### Actual behavior

On the docs site, https://opensource.adobe.com/spectrum-web-components/components/menu/, MenuItem seems to fail to initialize, and the appropriate `role` and state properties are not present.

### How can we reproduce this issue?

1. Go to https://opensource.adobe.com/spectrum-web-components/components/menu/
2. Inspect any of the `<sp-menuitem>` elements on the page using the browser's Web Developer tools.
3. Notice that none of the `<sp-menuitem>` elements have `role`, `aria-checked`, `aria-haspopup`, or `aria-expanded`.
4. Interestingly, when the `<sp-menu>` appears in a popover as with https://opensource.adobe.com/spectrum-web-components/components/action-menu/ or https://opensource.adobe.com/spectrum-web-components/components/picker/, the  `<sp-menuitem>` elements do seem to initialize with `role` and `aria-*` state attributes, and the `<sp-menu>` appears to initialize in Storybook stories.
5. When an `<sp-menuitem>` has a submenu, the parent `<sp-menuitem>` does not have `aria-haspopup` or `aria-expanded` to communicate the state of the submenu

**Comment by najikahalsema** [comment]
@majornista Hi! If you investigate [here](https://opensource.adobe.com/spectrum-web-components/storybook/?path=/story/menu--single-select), can you see the missing `aria-checked` and `role` attributes? I see them on my end for both the element in the popover and the element with no popover. 

However, the menu items without a submenu are definitely missing the right ARIA attributes. Just wanted to make sure, since there's #3248 for that problem only and we could close that issue if the other pieces of it are resolved.

**Comment by Westbrook** [comment]
I'm also no longer seeing this issue on the page in question. Closing for now. Please feel free to reopen with expanded reproduction steps if you feel like we're missing something here.
